src/app/pages/Collectibles/CollectibleItem.tsx

```diff
@@ -1,45 +1,81 @@
+
 interface Props {
   assetSlug: string;
-  index: number;
-  itemsLength: number;
+  accountPkh: string;
+  detailsShown: boolean;
 }
 
-export const CollectibleItem: FC<Props> = ({ assetSlug, index, itemsLength }) => {
+export const CollectibleItem: FC<Props> = ({ assetSlug, accountPkh, detailsShown }) => {
+  const { popup } = useAppEnv();
   const metadata = useAssetMetadata(assetSlug);
   const toDisplayRef = useRef<HTMLDivElement>(null);
   const [displayed, setDisplayed] = useState(true);
+  const { data: balance } = useBalance(assetSlug, accountPkh, { displayed });
+  const details = useOneCollectibleDetailsSelector(assetSlug);
+
+  const floorPrice = details?.floorPrice;
 
   const handleIntersection = useCallback(() => {
     setDisplayed(true);
   }, [setDisplayed]);
 
   useIntersectionDetection(toDisplayRef, handleIntersection, !displayed);
 
-  if (metadata == null) return null;
+  const assetName = getAssetName(metadata);
 
   return (
-    <Link to={`/collectible/${assetSlug}`}>
-      <div className="flex items-center" style={index === itemsLength - 1 ? {} : { borderBottom: '1px solid #e2e8f0' }}>
-        <div className="p-2">
-          <div
-            ref={toDisplayRef}
-            style={{ borderRadius: '12px' }}
-            className="border border-gray-300 w-16 h-16 flex items-center justify-center overflow-hidden p-2"
-          >
-            {displayed && <AssetIcon assetSlug={assetSlug} className="w-12 h-12" />}
+    <Link to={`/collectible/${assetSlug}`} className="flex flex-col border border-gray-300 rounded-lg">
+      <div
+        ref={toDisplayRef}
+        className={clsx(
+          'relative flex items-center justify-center bg-blue-50 rounded-lg overflow-hidden hover:opacity-70',
+          detailsShown ? 'border-b border-gray-300' : undefined
+        )}
+        title={assetName}
+        style={{ height: popup ? 106 : 125 }}
+      >
+        {displayed && <CollectibleItemImage metadata={metadata} assetSlug={assetSlug} />}
+
+        {balance ? (
+          <div className="absolute bottom-1.5 left-1.5 text-2xs text-white leading-none p-1 bg-black bg-opacity-60 rounded">
+            {balance.toFixed()}×
           </div>
-        </div>
-        <div className="pl-2">
-          <p className="text-sm text-gray-910">{getAssetName(metadata)}</p>
-        </div>
+        ) : null}
       </div>
+
+      {detailsShown && (
+        <div className="mt-1 mb-2 mx-1.5">
+          <h5 className="text-sm leading-5 text-gray-910 truncate">{assetName}</h5>
+          <div className="text-2xs leading-3 text-gray-600">
+            <span>Floor: </span>
+            {isDefined(floorPrice) ? (
+              <>
+                <Money shortened smallFractionFont={false} tooltip={true} cryptoDecimals={TEZOS_METADATA.decimals}>
+                  {atomsToTokens(floorPrice, TEZOS_METADATA.decimals)}
+                </Money>
+                <span> TEZ</span>
```

Comment:
Why do we hardcode `TEZ` symbol? From objkt we can get also `uUSD` and `USDtz` currencies.